src/Compositor.cpp

```diff
@@ -3039,31 +3040,39 @@ void CCompositor::ensurePersistentWorkspacesPresent(const std::vector<SWorkspace
             continue;
         }
 
-        WORKSPACEID id     = rule.workspaceId;
-        std::string wsname = rule.workspaceName;
-        if (id == WORKSPACE_INVALID) {
-            const auto R = getWorkspaceIDNameFromString(rule.workspaceString);
-            id           = R.id;
-            wsname       = R.name;
-        }
+        auto PWORKSPACE = pWorkspace;
```

Comment:
I don't think this is correct?

Let's say we want to recheck Workspace `1`, so pWorkspace is `1`

Then, we have a rule
```
workspace = 2, persistent:1, monitor:DP-2
```

we get to this spot with our rule, but set `1`'s persistent to true...

Reply:
I am calling this method passing a single rule that i already derived from the workspace from Workspace.cpp:651 'const auto WORKSPACERULE = g_pConfigManager->getWorkspaceRuleFor(m_pSelf.lock());'. 
So i'm already sure that the workspace match the rule, but I agree with you that the method signature could be misleading without this info. I'll add a check to only apply the matching rules.